src/pymodaq_gui/parameter/utils.py

```diff
@@ -166,14 +166,33 @@ def iter_children(param, childlist=[]):
     return childlist
 
 
-def iter_children_params(param, childlist=[]):
+def iter_children_params(param, childlist=[], filter_type=[], filter_name=[], selectFilter=False):
     """Get a list of parameters under a given Parameter
+        | Returns all children parameters.
 
+        =============== ================================= ====================================
+        **Parameters**   **Type**                           **Description**
+        *param*          instance of pyqtgraph parameter    the root node to be coursed
+        *childlist*      list                               the child list recetion structure
+        *filter_type*    list                               filter parameter sharing those types from output
+        *filter_name*    list                               filter parameter sharing those names from output
+        *selectFilter*   bool                               if True, add filtered parameters to output list
+
+        =============== ================================= ====================================
+
+        Returns
+        -------
+        childlist : parameter list
+            The list of the children from the given node.
     """
     for child in param.children():
-        childlist.append(child)
+        isFiltered = child.type() in filter_type or child.name() in filter_name
+        addSelectChild = selectFilter and isFiltered
+        addNonSelectChild = not selectFilter and not isFiltered
+        if addSelectChild or addNonSelectChild:
+            childlist.append(child)
         if child.hasChildren():
-            childlist.extend(iter_children_params(child, []))
```

Comment:
I did this because using list in arguments can cause some issues. I know it seems fine here but I wonder if it can cause some hard to see issues